x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsqlExpressionTranslators.java

```diff
@@ -118,10 +121,52 @@ public static Query doTranslate(BinaryComparison bc, TranslatorHandler handler)
             ExpressionTranslators.BinaryComparisons.checkBinaryComparison(bc);
             Query translated = translateOutOfRangeComparisons(bc);
             return translated == null
-                ? ExpressionTranslators.BinaryComparisons.doTranslate(bc, handler)
+                ? ExpressionTranslators.BinaryComparisons.doTranslate(bc, handler, BinaryComparisons::dispatchEsqlBinaryComparisons)
                 : handler.wrapFunctionQuery(bc, bc.left(), () -> translated);
         }
 
+        public static Query dispatchEsqlBinaryComparisons(
+            BinaryComparison bc,
+            TypedAttribute attribute,
+            Source source,
+            String name,
+            Object value,
+            String format,
+            boolean isDateLiteralComparison,
+            ZoneId zoneId
+        ) {
+            if (bc instanceof org.elasticsearch.xpack.esql.evaluator.predicate.operator.comparison.GreaterThan) {
```

Comment:
I didn't review the usages of the QL `BinaryComparison` classes in this file, but it might be the case that we can/should also replace those usages with their ES|QL counterparts, in which case we could just import these classes directly and not need a fully qualified class name here.